js_modules/dagster-ui/packages/ui-core/src/assets/AssetsCatalogTable.tsx

```diff
@@ -26,46 +26,96 @@ import {FIFTEEN_SECONDS, useRefreshAtInterval} from '../app/QueryRefresh';
 import {currentPageAtom} from '../app/analytics';
 import {PythonErrorFragment} from '../app/types/PythonErrorFragment.types';
 import {AssetGroupSelector} from '../graphql/types';
+import {useUpdatingRef} from '../hooks/useUpdatingRef';
 import {PageLoadTrace} from '../performance';
 import {useBlockTraceUntilTrue} from '../performance/TraceContext';
-import {useIndexedDBCachedQuery} from '../search/useIndexedDBCachedQuery';
+import {fetchPaginatedData} from '../runs/fetchPaginatedBucketData';
+import {CacheManager} from '../search/useIndexedDBCachedQuery';
 import {LoadingSpinner} from '../ui/Loading';
 
 type Asset = AssetTableFragment;
 
 const groupTableCache = new Map();
 
+const ASSET_CATALOG_TABLE_QUERY_VERSION = 1;
+const BATCH_LIMIT = 100000;
+
 export function useAllAssets(groupSelector?: AssetGroupSelector) {
   const client = useApolloClient();
   const [{error, assets}, setErrorAndAssets] = useState<{
     error: PythonErrorFragment | undefined;
     assets: Asset[] | undefined;
   }>({error: undefined, assets: undefined});
 
-  const {localCacheIdPrefix} = useContext(AppContext);
+  const assetsRef = useUpdatingRef(assets);
 
-  const assetsQuery = useIndexedDBCachedQuery<
-    AssetCatalogTableQuery,
-    AssetCatalogTableQueryVariables
-  >({
-    key: `${localCacheIdPrefix}/allAssets`,
-    query: ASSET_CATALOG_TABLE_QUERY,
-    version: 1,
-  });
-  // Delete old database from before the prefix, remove this at some point
-  indexedDB.deleteDatabase('indexdbQueryCache:allAssets');
+  const {localCacheIdPrefix} = useContext(AppContext);
 
-  const {data, fetch: fetchAssets} = assetsQuery;
+  const cacheManager = useMemo(
+    () => new CacheManager<AssetTableFragment[]>(`${localCacheIdPrefix}/allAssetNodes`),
+    [localCacheIdPrefix],
+  );
 
-  const assetsOrError = data?.assetsOrError;
   useLayoutEffect(() => {
-    if (assetsOrError) {
-      setErrorAndAssets({
-        error: assetsOrError?.__typename === 'PythonError' ? assetsOrError : undefined,
-        assets: assetsOrError?.__typename === 'AssetConnection' ? assetsOrError.nodes : undefined,
+    cacheManager.get(ASSET_CATALOG_TABLE_QUERY_VERSION).then((data) => {
+      if (data && !assetsRef.current) {
+        setErrorAndAssets({
+          error: undefined,
+          assets: data,
+        });
+      }
+    });
+  }, [cacheManager, assetsRef]);
+
+  const fetchAssets = useCallback(async () => {
+    try {
+      const data = await fetchPaginatedData({
+        async fetchData(cursor: string | undefined) {
```

Comment:
Not urgent but while we're in here, could we make fetchData not use  `dataSoFar.push(...data)`? Technically there's a limit to the number of objects we can fetch at a time if we use the spread operator to pass them each as an argument. In JavaScriptCore the limit is `65536` but I think it's higher in Chrome?

If our page size is 100k we could def get close to the max there

Reply:
edit: Actually it is urgent, turns out you can't `array.push(...100k items)` in Safari at all

<img width="414" alt="image" src="https://github.com/dagster-io/dagster/assets/1037212/ec64bf57-c578-4887-ba38-644341da582e">

Reply:
Ah right I remember we ran into this before.

Reply:
Though we're lowering the limit to 1,000 now 😬